monai/apps/pathology/transforms/post/array.py

```diff
@@ -0,0 +1,329 @@
+
+    def __call__(  # type: ignore
+        self, image: NdarrayOrTensor, mask: Optional[NdarrayOrTensor] = None, markers: Optional[NdarrayOrTensor] = None
+    ) -> NdarrayOrTensor:
+        """
+        Args:
+            image: image where the lowest value points are labeled first. Shape must be [1, H, W, [D]].
+            mask: optional, the same shape as image. Only points at which mask == True will be labeled.
+                If None (no mask given), it is a volume of all 1s.
+            markers: optional, the same shape as image. The desired number of markers, or an array marking
+                the basins with the values to be assigned in the label matrix. Zero means not a marker.
+                If None (no markers given), the local minima of the image are used as markers.
+        """
+
+        image = convert_to_numpy(image)
+        markers = convert_to_numpy(markers)
+        mask = convert_to_numpy(mask)
+
+        instance_seg = watershed(image, markers=markers, mask=mask, connectivity=self.connectivity)
+
+        return convert_to_dst_type(instance_seg, image, dtype=self.dtype)[0]
+
+
+class GenerateMask(Transform):
+    """
+    generate mask used in `watershed`. Only points at which mask == True will be labeled.
+
+    Args:
+        softmax: if True, apply a softmax function to the prediction.
+        sigmoid: if True, apply a sigmoid function to the prediction.
+        threshold: if not None, threshold the float values to int number 0 or 1 with specified theashold.
+        remove_small_objects: whether need to remove some objects in the marker. Defaults to True.
+        min_size: objects smaller than this size are removed if `remove_small_objects` is True. Defaults to 10.
+        dtype: target data content type to convert, default is np.uint8.
+
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        softmax: bool = True,
+        sigmoid: bool = False,
+        threshold: Optional[float] = None,
+        remove_small_objects: bool = True,
+        min_size: int = 10,
+        dtype: DtypeLike = np.uint8,
+    ) -> None:
+        if sigmoid and threshold is None:
+            raise ValueError("Threshold is needed when using sigmoid activation.")
+
+        self.dtype = dtype
+        self.activations = Activations(sigmoid=sigmoid, softmax=softmax)
+        self.asdiscrete = AsDiscrete(threshold=threshold, argmax=softmax)
+        if remove_small_objects:
+            self.remove_small_objects = RemoveSmallObjects(min_size=min_size)
+        else:
+            self.remove_small_objects = None  # type: ignore
+
+    def __call__(self, prob_map: NdarrayOrTensor) -> NdarrayOrTensor:
+        """
+        Args:
+            prob_map: probability map of segmentation, shape must be [C, H, W, [D]]
+        """
+
+        pred = self.activations(prob_map)
+        pred = self.asdiscrete(pred)
+
+        if isinstance(pred, torch.Tensor):
+            pred = pred.detach().cpu()
```

Comment:
Maybe use `pred = convert_to_numpy(pred)` directly?

Thanks.

Reply:
Have updated it , thx!